### PR TITLE
Update: Deprecate discovery mode

### PR DESCRIPTION
**Pull request by ericrpowers**
#### Description
We will be moving away from discovery mode in favor of leveraging the namespace property `EnforcerDefaultBehavior` in conjunction with enforcer changes to create similar behavior.

### CODE REVIEW

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "deprecate-discoverymode",
    "component": "backend",
    "pr-id": "946",
    "commit-sha": "d854ca8a9813279906391d52f36b6c046cc6b103",
    "pipeline": "master"
  },
  {
    "project": "deprecate-discoverymode",
    "component": "gaia",
    "pr-id": "580",
    "commit-sha": "0506cb6963995c5ef45693ff4ab0101eb58c32a3",
    "pipeline": "master"
  }
]
```

File: specs/root.spec [hunk 1]
```diff
@@ -248,9 +248,10 @@ relations:
 
 - rest_name: discoverymode
   get:
-    description: Returns the list of discovery modes.
+    description: (Deprecated) Returns the list of discovery modes.
   create:
-    description: Deploy the discovery mode assets onto the specified namespace.
+    description: (Deprecated) Deploy the discovery mode assets onto the specified
```
Comment (t00f) [inline specs/root.spec:253]: There should be a deprecated option here.

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "deprecate-discoverymode",
    "component": "backend",
    "pr-id": "946",
    "commit-sha": "d854ca8a9813279906391d52f36b6c046cc6b103",
    "pipeline": "master"
  },
  {
    "project": "deprecate-discoverymode",
    "component": "gaia",
    "pr-id": "580",
    "commit-sha": "512115c5c39b9363a37e4fb8bbe51625f11a28d2",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "deprecate-discoverymode",
    "component": "backend",
    "pr-id": "946",
    "commit-sha": "a848df2d48ef33eb913225833f0b67a79548a939",
    "pipeline": "master"
  },
  {
    "project": "deprecate-discoverymode",
    "component": "gaia",
    "pr-id": "580",
    "commit-sha": "fb3f46da8ffa08cc807d9e31d58eecd2de83eea5",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "deprecate-discoverymode",
    "component": "backend",
    "pr-id": "946",
    "commit-sha": "19a2fccdd2e6e4f3b9112fc7c012750cca64518b",
    "pipeline": "master"
  },
  {
    "project": "deprecate-discoverymode",
    "component": "gaia",
    "pr-id": "580",
    "commit-sha": "fb3f46da8ffa08cc807d9e31d58eecd2de83eea5",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "deprecate-discoverymode",
    "component": "backend",
    "pr-id": "946",
    "commit-sha": "19a2fccdd2e6e4f3b9112fc7c012750cca64518b",
    "pipeline": "master"
  },
  {
    "project": "deprecate-discoverymode",
    "component": "gaia",
    "pr-id": "580",
    "commit-sha": "fb3f46da8ffa08cc807d9e31d58eecd2de83eea5",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "deprecate-discoverymode",
    "component": "backend",
    "pr-id": "946",
    "commit-sha": "6bede431d877740aea1117e2dc0df50415500018",
    "pipeline": "master"
  },
  {
    "project": "deprecate-discoverymode",
    "component": "gaia",
    "pr-id": "580",
    "commit-sha": "65f50a5bbae8fb36eedff9ddc14ede8d22b65a45",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "deprecate-discoverymode",
    "component": "backend",
    "pr-id": "946",
    "commit-sha": "97deefb83093f394b1af8caee521c2abee5b05fb",
    "pipeline": "master"
  },
  {
    "project": "deprecate-discoverymode",
    "component": "gaia",
    "pr-id": "580",
    "commit-sha": "a0c14e7253c0107d7304e9479b6052d0aa9557f6",
    "pipeline": "master"
  }
]
```

**Review by t00f — APPROVED** [review]

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "deprecate-discoverymode",
    "component": "backend",
    "pr-id": "946",
    "commit-sha": "97deefb83093f394b1af8caee521c2abee5b05fb",
    "pipeline": "master"
  },
  {
    "project": "deprecate-discoverymode",
    "component": "gaia",
    "pr-id": "580",
    "commit-sha": "a0c14e7253c0107d7304e9479b6052d0aa9557f6",
    "pipeline": "master"
  }
]
```

**Pull request merged**